### DISCUSSION TITLE
DataTable - Automatic cell selection with Keyboard

### DISCUSSION THREAD

**Discussion by ConradMcLaughlin** [primefaces/community#370] [PrimeReact]
If I configure the DataTable to use cellSelection, I can navigate between rows by pressing the up and down arrows on the keyboard. To select an actual row however I have to press the spacebar.

Is there a way that I can automatically select the highlighted row after the keypress of up or down? Without having to press the spacebar?

**Comment by ConradMcLaughlin** [comment]
I have achieved this myself in the end by creating an EventHandler on the DataTable for _onKeyUp_ (which works if the DataTable is configured in a particular way).

**Resolution summary**
1. I conditionally set the selectionMode of my checkbox column to _{null}_ when I want to enable preview. This allows for keyboard navigation.
2. However, I have to use 'spacebar' or 'Enter' to select the row that the keyboard has hovered over. To avoid this and have automatic selection I have created this as the event handler:
`onKeyUp={(e) => {
    if (e.key === 'ArrowUp' || e.key === 'ArrowDown') { 
        const focusedCell = e.target;
        focusedCell.click();
    }
}}`

Note: I did not need to enable the cellSelection setting in the end (for my scenario).  The selectionMode of my DataTable remains as 'checkbox'.

**Thoughts**
- I could not see where the use of the 'onKeyUp' event is documented. 
- I think it is worth exposing a new event called onKeyboardHover (or something of the nature) as a wrapper for this. Which could return the same data object used for the other DataTable events.
- Also I think it may be worth having a setting on the DataTable itself which allows you to autoselect rows when using the keyboard.

I'll look to suggest this as an idea (if I can) elsewhere on this gitHub page.


Thanks

**Comment by mertsincan** [comment]
Hi,

Maybe, you can use `pt` prop for such cases. Exp;
```
pt="{
   cell: {
         onKeyUp: ....
         onKeyDown: ...
   }
}"
```

Best Regards,
